conda_recipe_manager/parser/dependency.py

```diff
@@ -0,0 +1,109 @@
+"""
+:Description: Provides types and utilities for managing recipe dependencies.
+"""
+
+from __future__ import annotations
+
+from enum import Enum, auto
+from typing import NamedTuple, Optional
+
+from conda.models.match_spec import MatchSpec
+
+from conda_recipe_manager.parser.selector_parser import SelectorParser
+from conda_recipe_manager.parser.types import SchemaVersion
+
+
+class DependencySection(Enum):
+    """
+    Enumerates dependency sections found in a recipe file.
+    """
+
+    BUILD = auto()
+    HOST = auto()
+    RUN = auto()
+    RUN_CONSTRAINTS = auto()
+    # NOTE: This was not in the `requirements/` section in V0
```

Comment:
Sorry to nitpick, but I think the placement of this comment doesn't indicate which requirements section wasn't in V0 (to a new person, they might be like "Is it the section above, or the section below?"); I would suggest either putting the comment on the same line or specify what the new section is directly in the comment.

Reply:
Usually a comment refers to the line directly below it. That being said, I think it would read better replacing `This`. Thanks for the call-out!